<launch>
  <!--Gazebo arguments-->
  <arg name="paused" default="false"/>
  <arg name="gui" default="true"/>

  <!--load signaturebot URDF to parameter server as robot_description-->
  <param name="robot_description" textfile="$(find fyp)/urdf/signaturebot_effort.urdf"/>
  <!--load arm_controller and joint_limits to parameter server-->
  <rosparam file="$(find fyp)/config/effort_controller.yaml" command="load"/>

  <!--launch empty_world in Gazebo with arguments-->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
   <arg name="world_name" default="worlds/empty.world"/>
   <arg name="paused" value="$(arg paused)"/>
   <arg name="gui" value="$(arg gui)"/>
 </include>

  <!--spawn controllers using respective controller managers-->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="/signaturebot"
    args="
    arm_controller/state
    arm_controller/effort/pitch_joint
    arm_controller/effort/yaw_joint
    arm_controller/effort/extension_joint
    "/>

  <!--publish joint states to /tf for rviz etc-->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
    output="screen">
  </node>

  <!--spawn robot_description into Gazebo from parameter server-->
  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model signaturebot"/>

  <!--general joystick interface that publishes to /sensor_msgs/joy-->
  <node pkg="joy" type="joy_node" name="joy">
    <param name="deadzone" value="0.2" />
    <param name="autorepeat_rate" value="40" />
    <param name="coalesce_interval" value="0.025" />
  </node>

  <node pkg="fyp" name="force_position" type="force_position.py" />
  <node pkg="fyp" name="active_constraint_demo" type="elastic_constraint.py" output="screen" launch-prefix="gnome-terminal --command" />

  <node
    name="rviz"
    pkg="rviz"
    type="rviz"
    args="-d $(find fyp)/rviz_config/wrench_vis.rviz" required="true" />

</launch>
